internal/pkg/status/legacy_status.go

```diff
@@ -0,0 +1,465 @@
+		readyCondition.Reason = "Missing .status.desiredNumberScheduled"
+		return []Condition{readyCondition}, nil
+	}
+
+	if desiredNumberScheduled > currentNumberScheduled {
+		readyCondition.Reason = fmt.Sprintf("Waiting for desired replicas to be scheduled. Current: %d/%d", currentNumberScheduled, desiredNumberScheduled)
+		return []Condition{readyCondition}, nil
+	}
+
+	if desiredNumberScheduled > updatedNumberScheduled {
+		readyCondition.Reason = fmt.Sprintf("Waiting for updated replicas to be scheduled. Updated: %d/%d", updatedNumberScheduled, desiredNumberScheduled)
+		return []Condition{readyCondition}, nil
+	}
+
+	if desiredNumberScheduled > numberAvailable {
+		readyCondition.Reason = fmt.Sprintf("Waiting for replicas to be available. Available: %d/%d", numberAvailable, desiredNumberScheduled)
+		return []Condition{readyCondition}, nil
+	}
+
+	if desiredNumberScheduled > numberReady {
+		readyCondition.Reason = fmt.Sprintf("Waiting for replicas to be ready. Ready: %d/%d", numberReady, desiredNumberScheduled)
+		return []Condition{readyCondition}, nil
+	}
+
+	// All ok
+	readyCondition.Status = "True"
+	readyCondition.Reason = fmt.Sprintf("All replicas scheduled as expected. Replicas: %d", desiredNumberScheduled)
+	return []Condition{readyCondition}, nil
+}
+
+// PVC
+func pvcConditions(u *unstructured.Unstructured) ([]Condition, error) {
+	obj := u.UnstructuredContent()
+	readyCondition := NewFalseCondition(ConditionReady)
+	phase := clientu.GetStringField(obj, ".status.phase", "unknown")
+	if phase != "Bound" { // corev1.ClaimBound
+		readyCondition.Reason = fmt.Sprintf("PVC is not Bound. phase: %s", phase)
+		return []Condition{readyCondition}, nil
+	}
+	// All ok
+	readyCondition.Status = "True"
+	readyCondition.Reason = "PVC is Bound"
+	return []Condition{readyCondition}, nil
+}
+
+// Pod
+func podConditions(u *unstructured.Unstructured) ([]Condition, error) {
+	rv := []Condition{}
+	readyCondition := NewFalseCondition(ConditionReady)
+	obj := u.UnstructuredContent()
+
+	phase := clientu.GetStringField(obj, ".status.phase", "unknown")
+	objc := clientu.GetObjectWithConditions(obj)
+
+	for _, c := range objc.Status.Conditions {
+		switch c.Type {
+		case "Ready":
+			readyCondition.Reason = c.Reason
+			readyCondition.Status = c.Status
+			if c.Status == "False" {
+				if c.Reason == "PodCompleted" {
+					readyCondition.Status = "True"
+					if phase == "Succeeded" {
+						rv = append(rv, NewCondition(ConditionCompleted, "Pod Succeeded").Get())
+					} else {
+						rv = append(rv, NewCondition(ConditionFailed, fmt.Sprintf("Pod phase: %s", phase)).Get())
+					}
+				}
+			}
+		}
+	}
+
+	if readyCondition.Reason == "" {
+		readyCondition.Reason = "Phase: " + phase
+	} else {
+		readyCondition.Reason = "Phase: " + phase + ", " + readyCondition.Reason
+	}
+	rv = append(rv, readyCondition)
+	return rv, nil
+}
+
+// PodDisruptionBudget
+func pdbConditions(u *unstructured.Unstructured) ([]Condition, error) {
```

Comment:
I've seen this case when checking immediately that I wouldn't consider "Ready":

```
apiVersion: policy/v1beta1
kind: PodDisruptionBudget
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"policy/v1beta1","kind":"PodDisruptionBudget","metadata":{"annotations":{},"name":"test-app-pdb","namespace":"test-local-deploy-all"},"spec":{"minAvailable":2,"selector":{"matchLabels":{"app":"test-app-deployment"}}}}
  creationTimestamp: 2019-06-11T15:11:58Z
  generation: 1
  name: test-app-pdb
  namespace: test-local-deploy-all
  resourceVersion: "6036052"
  selfLink: /apis/policy/v1beta1/namespaces/test-local-deploy-all/poddisruptionbudgets/test-app-pdb
  uid: 41d83cf7-8c5b-11e9-8840-42010a8e00dc
spec:
  minAvailable: 2
  selector:
    matchLabels:
      app: test-app-deployment
status:
  currentHealthy: 0
  desiredHealthy: 0
  disruptionsAllowed: 0
  expectedPods: 0
  observedGeneration: 1
```

Does it make sense to make sure `metadata.minAvailable` == `status.desiredHealthy`?

Reply:
The check is if desiredHealthy > currentHealthy { Not Ready }
Isnt this sufficient.

Reply:
I believe in most cases that would be true. However in the case above (I ran `kubectl apply -f <file> -o yaml` to get the output immediately), `status.currentHealthy` == `status.desiredHealthy`, but `status.desiredHealthy` != `spec.minAvailable`, which I think is an indication that it is not ready yet.

Reply:
I'm not sure I'm following. What is % and what is count?

Reply:
iam checking for .currentHealth == 0 => not ready.

Reply:
What if the user passes in `minAvailable` as 0?